src/main/kotlin/de/livepoll/api/service/PollItemService.kt

```diff
@@ -143,21 +150,63 @@ class PollItemService {
 
     //-------------------------------------------- Update --------------------------------------------------------------
 
-    fun updateMultipleChoiceItem(pollItemId: Long, pollItem: MultipleChoiceItemDtoIn): MultipleChoiceItemDtoOut {
+    /**
+     * Move poll item to another position while updating the positions of other poll items in the poll.<br>
+     *
+     * This method works in-place and will adjust the poll items list.<br>
+     * Old and new position are counted from 1 onwards, NOT from 0!
+     */
+    fun movePollItem(oldPos: Int, newPos: Int, pollItems: MutableList<PollItem>) {
+        // Check if new position is existent
+        if (!(newPos >= 1 && newPos <= pollItems.size)) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid poll item position")
+        }
+
+        // Sort poll items by position instead of id
+        pollItems.sortBy { it.position }
+
+        // Update elements in between
+        when {
```

Comment:
I think a when is overkill here when you only have two options

Reply:
Yeah, true, IntelliJ was suggesting this 🤣